source/server/listener_manager_impl.h

```diff
@@ -127,6 +128,38 @@ struct ListenerManagerStats {
   ALL_LISTENER_MANAGER_STATS(GENERATE_COUNTER_STRUCT, GENERATE_GAUGE_STRUCT)
 };
 
+/**
+ * Provides the draining filter chains and the functionality to schedule listener destroy.
+ */
+class DrainingFilterChainsImpl {
```

Comment:
nit: since no longer an impl of an interface I would call this `DrainingFilterChainsManager` or something like that.

Reply:
I think it's fine to leave here. Up to you.

Reply:
Leave it here for now. It's the candidate at future refactor